experiment/simulation/js/main.js

```diff
@@ -1 +1,478 @@
+	const viewButton = document.getElementById('viewButton');
+	const playButton = document.getElementById('play');
+	const pauseButton = document.getElementById('pause');
+	const restartButton = document.getElementById('restart');
+
+	forceMenu.addEventListener('change', function() { window.clearTimeout(tmHandle); forceDirn = Number(forceMenu.value); restart(); });
+	viewButton.addEventListener('click', function() { window.clearTimeout(tmHandle); view = !view; restart(); });
+	pauseButton.addEventListener('click', function() { window.clearTimeout(tmHandle); });
+	playButton.addEventListener('click', function() {  window.clearTimeout(tmHandle); tmHandle = setTimeout(draw, 1000 / fps); });
+	restartButton.addEventListener('click', function() {restart();});
+
+	function init()
+	{
+		bldgTop = [
+			[startL[0] - 50, defY + thickness],
+			[startR[0], defY + 150 + thickness],
+			[startL[3], defY + 150 + thickness],
+			[startR[3] + 50, defY + thickness],
+		];
+
+		bldgTopLayer2 = [
+			[bldgTop[0][0], defY],
+			[bldgTop[0][0], defY + thickness],
+			[bldgTop[3][0], defY + thickness],
+			[bldgTop[3][0], defY],
+		];
+		bldgSide = [];
+
+		legs = [
+			[[startL[0], defY], [startL[0], defY - height]],
+			[[startL[1], defY], [startL[1], defY - 2 * height / 3]],
+			[[startR[2], defY], [startR[2], defY - 2 * height / 3]],
+			[[startR[3], defY], [startR[3], defY - height]],
+		];
+
+		if(view)
+		{
+			bldgTop[0][1] = defY - height;
+			bldgTop[1][0] = bldgTop[0][0];
+			bldgTop[2][0] = bldgTop[3][0];
+			bldgTop[3][1] = defY - height;
+
+			bldgTopLayer2 = [];
+			legs = [];
+		}
+
+		calc(mass, com, bldgTop);
+		calc(stiff, cor, bldgTop);
+		mid = [(bldgTop[0][0] + bldgTop[3][0]) / 2, (bldgTop[0][1] + bldgTop[1][1]) / 2];
+		dirn = 0;
+
+		if(com[0] != mid[0] && forceDirn > 1)
+		{
+			dirn = 1;
+			if((com[0] < mid[0] && forceDirn === 2) || (com[0] > mid[0] && forceDirn === 3))
+			{
+				dirn = -1;
+			}
+		}
+
+		else if(com[1] != mid[1] && forceDirn <= 1)
+		{
+			dirn = 1;
+			if((com[1] < mid[1] && forceDirn === 0) || (com[1] > mid[1] && forceDirn === 1))
+			{
+				dirn = -1;
+			}
+		}
+
+		if(cor[0] != mid[0] && forceDirn > 1)
+		{
+			dirn = -1;
+			if((cor[0] < mid[0] && forceDirn === 2) || (cor[0] > mid[0] && forceDirn === 3))
+			{
+				dirn = 1;
+			}
+		}
+
+		else if(cor[1] != mid[1] && forceDirn <= 1)
+		{
+			dirn = -1;
+			if((cor[1] < mid[1] && forceDirn === 0) || (cor[1] > mid[1] && forceDirn === 1))
+			{
+				dirn = 1;
+			}
+		}
+
+	}
+
+	function restart() 
+	{ 
+		window.clearTimeout(tmHandle); 
+		init();
+		tmHandle = window.setTimeout(draw, 1000 / fps); 
+	}
+
+	const slider_mass1 = document.getElementById("mass1");
+	const output_mass1 = document.getElementById("demo_mass1");
+	output_mass1.innerHTML = slider_mass1.value; // Display the default slider value
+
+	slider_mass1.oninput = function() {
+		output_mass1.innerHTML = this.value;
+		mass[0] = Number(document.getElementById("mass1").value);
+		restart();
+	};
```

Comment:
The code between lines 110-118 has been used multiple times to set the values of many elements. is it possible to refactor it in a function that takes the element ids as parameters?